src/index_main.cpp

```diff
@@ -3,10 +3,11 @@
 void setup_index_subcommand(CLI::App& app)
 {
     auto opt = std::make_shared<IndexOptions>();
-    auto index_subcmd = app.add_subcommand(
+    auto* index_subcmd = app.add_subcommand(
```

Comment:
is this a typo?

Reply:
No. It came up as a clang-tidy recommendation :shrug:   
Would this be an issue? I don't really understand the difference tbh

Reply:
Upon further inspection, this is fine, I was just noting this as this is not done in the other CLI code, but it is fine. This is actually more informative, as we know the variable is indeed a pointer!